test/functional/keyed_items.js

```diff
@@ -169,6 +169,23 @@ describe('rendering with keys', () => {
     expect(firstNode.parentNode).to.equal(null);
   });
 
+  it('should not throw error if rendering content on parent with outdated key map', () => {
+    function render(tag) {
+      elementVoid(tag, 'key');
+    }
+
+    patch(container, render, 'div');
+    const firstNode = container.childNodes[0];
+
+    container.removeChild(container.childNodes[0]);
+
+    expect(() => patch(container, render, 'span')).to.not.throw(Error);
```

Comment:
Yes, the test would fail, but it wouldn't be so clear that this is the main purpose of the test, making sure that this part of the code won't throw an error due to the element having been removed. I personally like to be more explicit about these things in my tests, but I can remove it if you prefer otherwise :)